File: backend/payment-api/src/domain/events/payment/payment-paid.event.ts
import { PaymentStatus } from 'src/domain/entities/payment.entity';
import { Event } from '../event';

export type PaymentPaidPayload = {
  id: string;
  paymentId: string;
  status: PaymentStatus.APPROVED;
};

export class PaymentPaidEvent extends Event {
  private constructor() {
    super('PaymentPaidEvent');
  }

  public static create(): PaymentPaidEvent {
    return new PaymentPaidEvent();
  }
}

File: backend/checkout-api/app/src/main/java/develcode/checkout/domain/events/PaymentPaidEvent.java
package develcode.checkout.domain.events;

import develcode.checkout.domain.shared.events.Event;

public class PaymentPaidEvent extends Event {

    private PaymentPaidEvent() {
        super("PaymentPaidEvent");
    }

    public static PaymentPaidEvent create() {
        return new PaymentPaidEvent();
    }

    public static PaymentPaidEvent createWith(final Object aPayload) {
        var event = new PaymentPaidEvent();
        event.setPayload(aPayload);
        return event;
    }

}
